bounties_api/std_bounties/tests.py

```diff
@@ -281,6 +387,71 @@ def test_activate_bounty(self):
             pk=self.bounty_to_activate_id)
         self.assertEqual(result, activated_bounty_from_db)
 
+    def test_fulfill_bounty(self):
+        bounty_to_fulfill = Bounty.objects.get(pk=self.bounty_to_fulfill_id)
+        fulfillment_timestamp = '1517536922'
+        fulfillment_datetime = datetime.fromtimestamp(int(fulfillment_timestamp))
+        fulfillment_id = 10
+        issuer = '0x4242424242424242424242424242424242424242'
+        inputs = {
+            'data': 'QmQjchBM6tjAvXzkDEpWgLUv9Ui4jwqtxsEzB6LxB2WqFL'
+        }
+        fulfillment = self.client.fulfill_bounty(
+            bounty=bounty_to_fulfill,
+            fulfillment_id=fulfillment_id,
+            inputs=inputs,
+            event_timestamp=fulfillment_timestamp,
+            transaction_issuer=issuer)
+
+        self.assertEqual(fulfillment.fulfillment_id, fulfillment_id)
+        self.assertEqual(fulfillment.bounty.id, self.bounty_to_fulfill_id)
+        self.assertEqual(fulfillment.accepted, False)
+        self.assertEqual(fulfillment.fulfiller, issuer)
+        self.assertEqual(fulfillment.fulfillment_created, fulfillment_datetime)
+
+        # Try fulfill with duplicate id
+        fulfillment = self.client.fulfill_bounty(
+            bounty=bounty_to_fulfill,
+            fulfillment_id=fulfillment_id,
+            inputs=inputs,
+            event_timestamp=fulfillment_timestamp,
+            transaction_issuer=issuer)
+        self.assertIsNone(fulfillment)
+
+    def test_accept_fulfillment(self):
+        bounty_to_fulfill = Bounty.objects.get(pk=self.bounty_to_fulfill_id)
```

Comment:
I know we already use this pattern of creating the data first, and then checking against it later. 

Do you think it would be more clear if each testing function was handling its own data creation? As this file grows, it is adding more and more cognitive load to keep track of what is being created and what is being tested. 

If you agree, could you move the bounty creation from https://github.com/Bounties-Network/BountiesAPI/pull/233/files#diff-da3568a7443e809355d1f23cc9695427R269 down to this function, and do the same for the other tests?

Reply:
Agree. I should've do this from the beginning.